File: CS/ShowToolTips/TypeConverters.cs
using DevExpress.Xpf.Map;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Globalization;
using System.Linq;
using System.Text;
using System.Windows.Data;

namespace ShowToolTips {
    class MapItemPopulationAttributeToStringTypeConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            ShapeTitle title = value as ShapeTitle;
            MapItem item;
            if (title == null)
                item = value as MapItem;
            else
                item = title.MapShape;

            if (item == null) return null;

            var attr = item.Attributes["POP_EST"];
            if (attr == null) return null;

            return attr.Value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }

    class MapItemGdpAttributeToStringTypeConverter : IValueConverter {
        public object Convert(object value, Type targetType, object parameter, CultureInfo culture) {
            ShapeTitle title = value as ShapeTitle;
            MapItem item;
            if (title == null)
                item = value as MapItem;
            else
                item = title.MapShape;

            if (item == null) return null;

            var attr = item.Attributes["GDP_MD_EST"];
            if (attr == null) return null;

            return attr.Value;
        }

        public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) {
            throw new NotImplementedException();
        }
    }
}

File: VB/ShowToolTips/TypeConverters.vb
Imports DevExpress.Xpf.Map
Imports System
Imports System.ComponentModel
Imports System.Globalization
Imports System.Windows.Data

Namespace ShowToolTips

    Friend Class MapItemPopulationAttributeToStringTypeConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim title As ShapeTitle = TryCast(value, ShapeTitle)
            Dim item As MapItem
            If title Is Nothing Then
                item = TryCast(value, MapItem)
            Else
                item = title.MapShape
            End If

            If item Is Nothing Then Return Nothing
            Dim attr = item.Attributes("POP_EST")
            If attr Is Nothing Then Return Nothing
            Return attr.Value
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class

    Friend Class MapItemGdpAttributeToStringTypeConverter
        Implements IValueConverter

        Public Function Convert(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.Convert
            Dim title As ShapeTitle = TryCast(value, ShapeTitle)
            Dim item As MapItem
            If title Is Nothing Then
                item = TryCast(value, MapItem)
            Else
                item = title.MapShape
            End If

            If item Is Nothing Then Return Nothing
            Dim attr = item.Attributes("GDP_MD_EST")
            If attr Is Nothing Then Return Nothing
            Return attr.Value
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
            Throw New NotImplementedException()
        End Function
    End Class
End Namespace
